src/EventHandler/AbstractMessage.php

```diff
@@ -68,18 +69,20 @@ public function __construct(
         array $info
     ) {
         parent::__construct($API);
-
-        $this->out = $rawMessage['out'];
-        $this->id = $rawMessage['id'];
-        $this->chatId = $info['bot_api_id'];
-        $this->senderId = isset($rawMessage['from_id'])
+        $decryptedMessage = $this instanceof SecretMessage ? $rawMessage['decrypted_message'] : null;
+        $this->out = $rawMessage['out'] ?? false;
+        $this->id = $this instanceof SecretMessage ? $rawMessage['random_id'] : $rawMessage['id'];
+        $this->chatId = $this instanceof SecretMessage ? $this->getClient()->getSecretChat($rawMessage['chat_id'])->chatId :$info['bot_api_id'];
+        $this->senderId = $this instanceof SecretMessage ? $this->getClient()->getSecretChat($rawMessage['chat_id'])->otherID : (isset($rawMessage['from_id'])
             ? $this->getClient()->getIdInternal($rawMessage['from_id'])
-            : $this->chatId;
+            : $this->chatId);
         $this->date = $rawMessage['date'];
-        $this->mentioned = $rawMessage['mentioned'];
-        $this->silent = $rawMessage['silent'];
-        $this->ttlPeriod = $rawMessage['ttl_period'] ?? null;
-
+        $this->mentioned = $rawMessage['mentioned'] ?? false;
+        $this->silent = $rawMessage['silent'] ?? $decryptedMessage['silent'];
+        $this->ttlPeriod = $rawMessage['ttl_period'] ?? $decryptedMessage['ttl'] ?? null;
+        if ($this instanceof SecretMessage && isset($decryptedMessage['reply_to_random_id'])) {
```

Comment:
Put this instead of $this->replyToMsgId = null; in the last else of this function